### DISCUSSION TITLE
Sonoff Pow Origin flashed with Tasmota dont start

### DISCUSSION THREAD

**Discussion by tronicman14** [arendst/Tasmota#18308] [Q&A] [answered]
Hello guys,
i have just flashed my new "Sonoff Pow Origin" with Tasmota, no problem to do it, but he dont want to start, i have wired AC220v to Lin and N , but i dont see any blink or light.

I have tried that configuration : 

`platformio.ini

[build_envs]
default_envs =  tasmota

[common]
framework                   = arduino
board                       = esp32dev #esp01_1m

`

`esptool.py --port /dev/ttyUSB0 --baud 115200 write_flash -fs 2MB -fm dout 0x0 Software/Tasmota-dev2/build_output/firmware/tasmota.bin
`

Can anyone help me please, i dont any idea :(

**Comment by sfromis** [comment, marked as answer]
Well, as a starting point, you need no build, but just follow the recommended flashing procedure of using the [Tasmota Web Installer](https://tasmota.github.io/install/), instead of inventing something which is **way** wrong. And when up, select the device in Auto-configuration.

If you build yourself, the esptool.py command should have no flash size parameter (especially not a wrong one), no flash mode, and use the right binary file, `tasmota32.factory.bin` (for initial flashing). But I see nothing in your configuration there making it even relevant to do your own build, and it does not look like you even read the guide about building Tasmota.

**Reply by tronicman14** [reply]
Hello sfromis and many thanks, i never tried the WebInstaller (its very usefull and simple) but i need to use USE_EXPRESSION and  SUPPORT_IF_STATEMENT, to use Conditional Rules i need to add some row at user_config_override.h and without compiling it i dont know how to do it

Many Thanks again and sorry for the late answer

**Reply by sfromis** [reply]
IME the ESP32 standard build already supports expressions and conditional rules. Did you try?

If you for some other reason need to build yourself, you need to select a build environment for ESP32 (like "tasmota32") instead of using one for ESP8266 (like "tasmota"). With a build for ESP32, the binary file name will start with "tasmota32" instead of just "tasmota".

Anyway, for ESP32, I can recommend looking at the much better programming features available with the Berry language, on this architecture, I'm not really using rules due to this.

**Reply by tronicman14** [reply]
Thanks for the advice .......... I immediately begin to read up on Berry Language

**Comment by MikeKulls12** [comment]
I had some issues with the Sonoff Pow Origin not starting after a successful flash via the web gui. Seeing this is the first hit I thought I would add my resolution. I believe the issue was related to the 3.3V supply not being strong enough. I had to use a second 3.3V supply from my power supply. Then the other issue is there are 2 possible bin files, tasmota32.bin and tasmota32.factory.bin. I believe it's the factory.bin that worked. Because I had to flash multiple times I'm not 100% sure it's the factory.bin but 99% sure.

**Reply by sfromis** [reply]
With ESP32 devices it is indeed **very** common that the USB/serial adapter is not workable to get on Wifi. You should disconnect 3.3V and connect mains power instead.

And yes, as explained in this thread, `tasmota32.factory.bin` is the one to use for wired flashing, but if using the recommended [Tasmota Web Installer](https://tasmota.github.io/install/), the right binary will be selected automatically.
